packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart

```diff
@@ -168,25 +170,35 @@ abstract class InAppPurchaseConnection {
   Future<bool> buyConsumable(
       {@required PurchaseParam purchaseParam, bool autoConsume = true});
 
-  /// (App Store only) Mark that purchased content has been delivered to the
+  /// Mark that purchased content has been delivered to the
   /// user.
   ///
   /// You are responsible for completing every [PurchaseDetails] whose
-  /// [PurchaseDetails.status] is [PurchaseStatus.purchased] or
-  /// [[PurchaseStatus.error]. Completing a [PurchaseStatus.pending] purchase
-  /// will cause an exception.
-  ///
-  /// This throws an [UnsupportedError] on Android.
-  Future<void> completePurchase(PurchaseDetails purchase);
+  /// [PurchaseDetails.status] is [PurchaseStatus.purchased].
+  /// Additionally, the purchase needs to be completed if the [PurchaseDetails.status]
+  /// [[PurchaseStatus.error].
+  /// Completing a [PurchaseStatus.pending] purchase will cause an exception.
+  /// For convenience, [PurchaseDetails.pendingCompletePurchase] indicates if a purchase is pending for completion.
+  ///
+  /// The method returns a [BillingResultWrapper] to indicate a detailed status of the complete process.
```

Comment:
This isn't totally clear about what `the complete process` is, do you mind explaining more? IE if the result says it failed, should the user try again? Is it possible for it to pending and the user should wait? That sort of thing.